ria_remote/create_sibling_ria.py

```diff
@@ -0,0 +1,305 @@
+
+        if sibling == storage_sibling:
+            # leads to unresolvable, circular dependency with publish-depends
+            raise ValueError("sibling names must not be equal")
+
+        # TODO: messages - this is "create-sibling". Don't confuse existence of local remotes with existence of the
+        #       actual remote sibling in wording
+        if not force and sibling in [r['name'] for r in ds.siblings(result_renderer=None)]:
+            yield get_status_dict(
+                status='error',
+                message="a sibling '{}' is already configured. Use --force to overwrite it.".format(sibling),
+                **res_kwargs,
+            )
+            return
+
+        if not force and storage_sibling in [r['name'] for r in ds.siblings(result_renderer=None)]:
+            yield get_status_dict(
+                status='error',
+                message="a storage-sibling '{}' is already configured. Use --force to overwrite it.".format(storage_sibling),
+                **res_kwargs,
+            )
+            return
+
+        # check special remote config:
+        # TODO: consider annexconfig the same way the special remote does (in-dataset special remote config)
+        base_path = ds.config.get("annex.ria-remote.{}.base-path".format(storage_sibling), None)
+        if not base_path:
+            yield get_status_dict(
+                status='impossible',
+                message="Missing required configuration 'annex.ria-remote.{}.base-path'".format(storage_sibling),
+                **res_kwargs,
+            )
+            return
+
+        base_path = Path(base_path)
+        ssh_host = ds.config.get("annex.ria-remote.{}.ssh-host".format(storage_sibling), None)
+        if not ssh_host:
+            lgr.warning("No SSH-Host configured for {}. Assume local RIA store at {}.".format(storage_sibling, base_path))
+        if ssh_host == '0':
+            ssh_host = None
+        lgr.info("create siblings '{}' and '{}' ...".format(sibling, storage_sibling))
+
+        lgr.debug('init special remote {}'.format(storage_sibling))
+        cmd = ['git', 'annex',
+               'initremote', storage_sibling,
+               'type=external',
+               'externaltype=ria',
+               'encryption=none',
+               'autoenable=true'
+               ]
+        result = subprocess.run(cmd, cwd=str(ds.path), stderr=subprocess.PIPE)
+        if result.returncode != 0:
+            if force and result.stderr.startswith(b'git-annex: There is already a special remote named'):
+                # run enableremote instead
+                cmd[2] = 'enableremote'
+                subprocess.run(cmd, cwd=str(ds.path))
+            else:
+                yield get_status_dict(
+                    status='error',
+                    message="initremote failed.\nstdout: %s\nstderr: %s" % (result.stdout, result.stderr),
+                    **res_kwargs
+                )
+                return
```

Comment:
Arrgh, we need https://github.com/datalad/datalad/issues/3839

But is it OK in the general case to use plain subprocess with `git annex` calls? Wouldn't we miss the entire environment setup logic? Why not use `AnnexRepo.init_remote()`?

Reply:
Done. However, not using `enable_remote` ATM, since it needs to take options first. Will file PR against core.